examples/tree_source/tree_source/app.py

```diff
@@ -1,119 +1,150 @@
+            self.load_children()
+        return self._children
+
+    def load_children(self):
         try:
-            decade_root = {
-                root.decade: root
-                for root in self._decades
-            }[decade]
-        except KeyError:
-            decade_root = Decade(decade)
-            self._decades.append(decade_root)
-            self._decades.sort(key=lambda v: v.decade)
-        movie = Movie(**entry)
-        decade_root._data.append(movie)
-        self._notify('insert', parent=decade_root, index=len(decade_root._data) - 1, item=movie)
+            names = os.scandir(self.path)
+            sub_paths = [os.path.join(self.path, p) for p in names if os.path.isdir(os.path.join(self.path, p))]
+            self._children = [Node(p, self) for p in sub_paths]
+        except OSError:
+            self._children = [LoadingFailedNode(self)]
+
+    def sort(self, accessor, key=None, reverse=False):
+
+        if accessor == "date_modified":  # use our own sort function
+            def sort_func(child):
+                return child._mtime
+        elif accessor == "name":
+            def sort_func(child):
+                return child.name[1].lower()
+        else:  # use the function provided by the user / default
+            def sort_func(child):
+                # sort according to value of accessor, using the provided sort key
+                try:
+                    attr = getattr(child, accessor)
+                    if isinstance(attr, tuple):
+                        icon, value = attr
+                    else:
+                        value = attr
+                    return key(value) if key else value
+                except AttributeError:
+                    return ''
+
+        # sort all children in hirarchy
+        self._children.sort(key=sort_func, reverse=reverse)
+
+        for c in self._children:
+            try:
+                c.sort(accessor, key, reverse)
+            except AttributeError:
+                pass
+
+
+class FileSystemSource(Node, Source):
+
+    def __init__(self, path):
+        super().__init__(path, parent=self)
+        self.path = path
+        self._parent = None
+        self._children = []
 
 
 class ExampleTreeSourceApp(toga.App):
-    # Table callback functions
-    def on_select_handler(self, widget, node):
-        if node and hasattr(node, 'title'):
-            self.label.text = 'You selected node: {}'.format(node.title)
-        else:
-            self.label.text = 'No row selected'
-
-    # Button callback functions
-    def insert_handler(self, widget, **kwargs):
-        entry = choice(bee_movies)
-        self.tree.data.add(entry)
 
     def startup(self):
         # Set up main window
         self.main_window = toga.MainWindow(title=self.name)
 
-        # Label to show responses.
-        self.label = toga.Label('Ready.')
+        root = os.path.abspath(os.sep)
+        self.fs_source = FileSystemSource(root)
```

Comment:
This won't work on Windows; might be better to use `pathlib.Path.home()` or `os.expanduser()`

Reply:
Done.